File: org.kevoree.modeling.microframework/src/main/java/org/kevoree/modeling/api/util/ActionType.java
package org.kevoree.modeling.api.util;

/**
 * Created by duke on 10/2/14.
 */
public enum ActionType {

    SET("S"),
    ADD("a"),
    REMOVE("r"),
    ADD_ALL("A"),
    REMOVE_ALL("R"),
    RENEW_INDEX("I"),
    CONTROL("C");

    private String code = "";

    ActionType(String code) {
        this.code = code;
    }

    public String toString() {
        return code;
    }

}

File: org.kevoree.modeling.microframework/src/main/java/org/kevoree/modeling/api/util/ActionType.kt
package org.kevoree.modeling.api.util

/**
 * Created with IntelliJ IDEA.
 * User: duke
 * Date: 05/08/13
 * Time: 19:15
 */

enum class ActionType(val code: String) {
    SET : ActionType("S")
    ADD : ActionType("a")
    REMOVE : ActionType("r")
    ADD_ALL : ActionType("A")
    REMOVE_ALL : ActionType("R")
    RENEW_INDEX : ActionType("I")
    CONTROL : ActionType("C")
}